python/TestHarness/unit_tests/test_HarnessTester.py

```diff
@@ -0,0 +1,74 @@
+import os
+import unittest
+import subprocess
+from TestHarnessTestCase import TestHarnessTestCase
+
+class TestHarnessTester(TestHarnessTestCase):
+  """
+  Test cyclic dependency error.
+  """
+  def testCyclic(self):
+    with self.assertRaises(subprocess.CalledProcessError) as cm:
+      self.runTests('-i', 'cyclic_tests')
+
+    e = cm.exception
+    self.assertEqual(e.returncode, 1)
+    self.assertIn('Cyclic or Invalid Dependency Detected!', e.output)
+    self.assertIn('tests/test_harness.testA', e.output)
+    self.assertIn('tests/test_harness.testB', e.output)
+    self.assertIn('tests/test_harness.testC', e.output)
+
+
+  """
+  Test skipping a test if its prereq is also skipped
+  """
+  def testDependencySkip(self):
```

Comment:
Comments for the method should be indented under the function definition, this allows the documentation to print if you use `help` from the python interpreter.